### PR TITLE
#168781842 travelled destination

### PR DESCRIPTION
**Pull request by Crispy-rw**
#### What does this PR do?
          This PR show all the destination and their visitors
#### Description of Task to be completed?
         This PR allows a user to view all the destinations with their visitors
#### How should this be manually tested?
         

- Clone this Branch

- run `npm i`

- run `npm db-migrate:dev`

- run `npm run dev`

> send a request a GET /api/v1/statistics/traveled-destinations

#### Any background context you want to provide?
        N/A
#### What are the relevant pivotal tracker stories?
        #168781842
#### Screenshots (if appropriate)
Screen Shot 2019-11-18 at 10 17 42

#### Questions:

### CODE REVIEW

File: src/routes/index.js [hunk 1]
```diff
@@ -14,5 +15,6 @@ app.use('/requests', requestRouter);
 app.use('/bookings', bookingRouter);
 app.use('/chats', chatRouter);
 app.use('/notifications', notificationRouter);
+app.use('/destinations', destinationRouter);
```
Comment (elchroy) [inline src/routes/index.js]: Let's change this to `/stats`

File: src/controllers/destinations.controller.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import sendResult from '../utils/sendResult';
+import DestinationsService from '../services/destination.service';
+
+
+const destinationCOntroller = {
```
Comment (elchroy) [inline src/controllers/destinations.controller.js]: `StatsController`, also change the filename to `stats.controller.js`

File: src/services/destination.service.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { Sequelize } from 'sequelize';
+import db from '../database/models/index';
+
+const DestinationsService = {
+  async getAllCities() {
```
Comment (elchroy) [inline src/services/destination.service.js]: `getTraveledDestinations`

**Review by urbanishimwe — APPROVED** [review]

File: src/controllers/statistics.controller.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import sendResult from '../utils/sendResult';
+import RequestsService from '../services/request.service';
+
+
+const StatisticsController = {
+  async getDestinations(req, res) {
+    const Travelled = await RequestsService.getTraveledDestinations();
```
Comment (elchroy) [inline src/controllers/statistics.controller.js]: let the variable name be `travelledDestination`

**Review by swaibat — APPROVED** [review]

**Review by alexismajya — APPROVED** [review]

File: src/services/request.service.js [hunk 5]
```diff
@@ -73,6 +74,13 @@ const RequestService = {
       where: { UserId: userId },
       include: { model: db.Trips, attributes: { exclude: ['RequestId'] } } });
     return result;
+  },
+  async getTraveledDestinations() {
+    // eslint-disable-next-line no-return-await
+    return await db.Trips.findAll({ attributes: ['country', 'city', [Sequelize.fn('COUNT', 'city'), 'N of visitors']],
+      group: ['city', 'country'],
+      order: [[Sequelize.fn('COUNT', 'city'), 'DESC']]
+    });
```
Comment (alexismajya) [inline src/services/request.service.js:92]: Great job @Crispy-lang. However, in this screenshot, the value on position `0`and value on position `3` has the same country and same city.

Screen Shot 2019-11-15 at 00 17 14

**Review by alexismajya — CHANGES_REQUESTED** [review]

File: src/services/request.service.js [hunk 6]
```diff
@@ -73,6 +74,13 @@ const RequestService = {
...
   }
```
Comment (alexismajya) [inline src/services/request.service.js:93]: I think we should also `exclude` the trips that belong to the requests that are `not yet approved`. or to be more specific, we can consider `done trips` only.

**Review by alexismajya — CHANGES_REQUESTED** [review]

File: src/services/request.service.js (hunk 6 above)
Comment (Crispy-rw) [reply src/services/request.service.js:93]: we considered done trips only

**Pull request merged**